exercises/practice/rest-api/.docs/instructions.md

```diff
@@ -0,0 +1,42 @@
+# Description
+
+Implement a RESTful API for tracking IOUs.
+
+Four roommates have a habit of borrowing money from each other frequently, and have trouble remembering who owes whom, and how much.
+
+Your task is to implement a simple [RESTful API](https://en.wikipedia.org/wiki/Representational_state_transfer) that receives [IOU](https://en.wikipedia.org/wiki/IOU)s as POST requests, and can deliver specified summary information via GET requests.
+
+## API Specification
+
+### User object
+
+```json
+{
+  "name": "Adam",
+  "owes": {
+    "Bob": 12.0,
+    "Chuck": 4.0,
+    "Dan": 9.5
+  },
+  "owed_by": {
+    "Bob": 6.5,
+    "Dan": 2.75
+  },
+  "balance": "<(total owed by other users) - (total owed to other users)>"
+}
+```
+
+### Methods
+
+| Description              | HTTP Method | URL    | Payload Format                                                            | Response w/o Payload                   | Response w/ Payload                                                             |
+| ------------------------ | ----------- | ------ | ------------------------------------------------------------------------- | -------------------------------------- | ------------------------------------------------------------------------------- |
+| List of user information | GET         | /users | `{"users":["Adam","Bob"]}`                                                | `{"users":<List of all User objects>}` | `{"users":<List of User objects for <users> (sorted by name)}`                  |
+| Create user              | POST        | /add   | `{"user":<name of new user (unique)>}`                                    | N/A                                    | `<User object for new user>`                                                    |
+| Create IOU               | POST        | /iou   | `{"lender":<name of lender>,"borrower":<name of borrower>,"amount":5.25}` | N/A                                    | `{"users":<updated User objects for <lender> and <borrower> (sorted by name)>}` |
+
+## Other Resources
+
+- [https://restfulapi.net/](https://restfulapi.net/)
+- Example RESTful APIs
+  - [GitHub](https://developer.github.com/v3/)
+  - [Reddit](https://www.reddit.com/dev/api/)
```

Comment:
General comment regarding the instructions: It seems you applied the suggestion from SleeplessByte to not have a payload (http request body) for the GET request. However, currently the student that would attempt this exercise would not know about this. I would recommend to fix this by adding an instructions.append.md file that explains how the payload is provided in case of the GET request.

Also there are other things that are currently not mentioned anywhere and should probably be clarified in the append file:
- no actual HTTP server should be implemented, just the handler functions should be written
- an in-memory object can serve as "database"
- the constructor function needs to accept a seed for the database as argument (and set up a sensible default if nothing was passed)

In addition is would probably be good to add the parameter names to the stub so the student understands what is provided as input (e.g. `post(url, payload)`).